app/uk/gov/hmrc/personaldetailsvalidation/RetryMongoIndexes.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.personaldetailsvalidation
+
+import reactivemongo.api.indexes.IndexType.Descending
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.{BSONDocument, BSONLong}
+import uk.gov.hmrc.mongo.ReactiveRepository
+
+import scala.collection.Seq
+import scala.concurrent.{ExecutionContext, Future}
+
+trait RetryMongoIndexes[A, B] { self: ReactiveRepository[A, B] =>
+
+  val ttlIndex = "personal-details-validation-ttl-index"
+  val OptExpireAfterSeconds = "expireAfterSeconds"
+  val createdAtField = "createdAt"
+  //user's CredId is the retry key
+  val retryKey = "credentialId"
+
+  val ttl: Long
+
+  override def ensureIndexes(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
+
+    import reactivemongo.bson.DefaultBSONHandlers._
+
+    val indexes: Future[List[Index]] = collection.indexesManager.list()
+
+    def ensureTtlIndex(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
```

Comment:
I am a little confused - why do both collections have the same indexes?  The new collection doesn't have "createdAt" field, right?

Reply:
But it *should* have a (unique) index on credentialId I would think?